### PR TITLE
Document setPath, startSharing, setTitle

### PR DESCRIPTION
**Pull request by paulproteus**
Requesting review from @zarvox and/or @mnutt.

### CODE REVIEW

**Comment by paulproteus** [comment]
The best way to review is probably to read the full docs:

* https://github.com/sandstorm-io/sandstorm/blob/fix-1198/docs/developing/http-apis.md

* https://github.com/sandstorm-io/sandstorm/blob/fix-1198/docs/developing/path.md

and then leave comments here.

File: docs/developing/http-apis.md [hunk 1]
```diff
@@ -169,6 +171,19 @@ automatically refreshes the IFRAME every 5 minutes.
...
+API requests. The API _client code_ can request that Sandstorm provide
+the IP address to the grain in an API request by setting a header:
+
+```
+X-Sandstorm-Passthrough: address
+```
+
+Your app can provide sample Javascript to make make API requests which
+sets this header on `XMLHttpRequest` calls to the API endpoint.
+
```
Comment (mnutt) [inline docs/developing/http-apis.md]: It'd be great to mention that the IP will be available to your backend through the legacy-http-bridge via the `X-Real-IP` header.

**Comment by paulproteus** [comment]
Self-merging because 100% docs

**Comment by paulproteus** [comment]
Thanks to @mnutt for asking for more things in this pull request; I attempted to do them.

@zarvox or others, if you have feedback, feel free to leave it here. 

https://docs.sandstorm.io/en/latest/developing/path/ is up to date now, etc.

File: docs/developing/path.md [hunk 2]
```diff
@@ -1,45 +1,117 @@
...
-
-The **easiest** way to handle Sandstorm's dynamic base path is to
-use the empty string (`''`) as your app's base path. This way,
-the app needs to make no decisions at runtime.
+### Recommendation: Use the empty string as your base URL
 
-Many web frameworks support this.
+The **easiest** way to handle Sandstorm's dynamic base URL is to use the empty
+string (`''`) as your app's base URL. This way, the app needs to make no
+decisions at runtime. Many web frameworks support this.
```
Comment (mnutt) [inline docs/developing/path.md]: When I first read this I was fairly sure I knew what it meant, but would have understood it clearer as "use URLs without hostnames" or even just "use relative URLs". I guess "base path" / "base url" are terms used in particular frameworks?

**Pull request merged**